controllers/component_build_controller.go

```diff
@@ -152,17 +160,66 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 	if val, ok := component.Annotations[gitops.PaCAnnotation]; ok && val == "1" {
 		// Use pipelines as code build
-		r.Log.Info("PaC enabled")
-		if err := r.EnsurePACRepository(ctx, component); err != nil {
+		r.Log.Info("Pipelines as Code enabled")
+
+		// Obtain Pipelines as Code callback URL
+		webhookTargetUrl, err := r.getPaCRoutePublicUrl(ctx)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		gitProvider, err := gitops.GetGitProvider(component)
+		if err != nil {
+			r.Log.Error(err, "error detecting git provider")
+			// Do not reconcile, because configuration must be fixed before it is possible to proceed.
+			return ctrl.Result{}, nil
+		}
+
+		// Expect that the secret contains token for Pipelines as Code webhook configuration,
+		// but under <git-provider>.token field. For example: github.token
+		// Also it can contain github-private-key and github-application-id
+		// in case GitHub Application is used instead of webhook.
+		pacSecret := corev1.Secret{}
+		if err := r.NonCachingClient.Get(ctx, types.NamespacedName{Namespace: pipelinesAsCodeNamespace, Name: gitopsprepare.PipelinesAsCodeSecretName}, &pacSecret); err != nil {
+			r.Log.Error(err, "failed to get Pipelines as Code secret")
 			return ctrl.Result{}, err
 		}
-		if err := r.GeneratePullRequest(ctx, component); err != nil {
+
+		if err := validatePaCConfiguration(gitProvider, pacSecret.Data); err != nil {
+			r.Log.Error(err, "Invalid configuration in Pipelines as Code secret")
+			// Do not reconcile, because configuration must be fixed before it is possible to proceed.
+			return ctrl.Result{}, nil
+		}
+
+		// Copy or update global PaC configuration in local namespace
+		if err := r.propagatePaCConfigurationSecretToLocalNamespace(ctx, req.Namespace, pacSecret); err != nil {
 			return ctrl.Result{}, err
 		}
 
+		// Generate webhook secret for the component git repository if not yet generated
+		// and stores it in the corresponding k8s secret.
+		webhookSecretString, err := r.ensureWebhookSecret(ctx, component)
+		if err != nil {
+			return ctrl.Result{}, err
```

Comment:
I'm assuming that errors here are expected to be intermittent, and simply returning and error and retrying makes sense here

Reply:
Yes, an error could occur because of network issue or because the resource was changed. Both are handled by a retry.